File: sunflower-core/src/main/java/org/tdf/sunflower/sync/GetAccounts.java
package org.tdf.sunflower.sync;

import org.tdf.common.util.HexBytes;

public class GetAccounts {
    private HexBytes stateRoot;
    private int maxAccounts;

    public GetAccounts(HexBytes stateRoot, int maxAccounts) {
        this.stateRoot = stateRoot;
        this.maxAccounts = maxAccounts;
    }

    public GetAccounts() {
    }

    public HexBytes getStateRoot() {
        return this.stateRoot;
    }

    public int getMaxAccounts() {
        return this.maxAccounts;
    }
}

File: sunflower-core/src/main/java/org/tdf/sunflower/sync/GetAccounts.kt
package org.tdf.sunflower.sync

import com.github.salpadding.rlpstream.annotation.RlpCreator
import com.github.salpadding.rlpstream.annotation.RlpProps
import org.tdf.common.util.HexBytes

@RlpProps("stateRoot", "maxAccounts")
data class GetAccounts @RlpCreator constructor(val stateRoot: HexBytes, val maxAccounts: Int)